### PR TITLE
refactor(share_plus)!: two-package federated architecture

### PR DESCRIPTION
**Pull request by nohli**
## Description

Refactor share_plus from the federated architecture to a platform interface and a package with all platform implementations.

## Related Issues

https://github.com/fluttercommunity/plus_plugins/issues/1226

## Checklist

- [x] I read the Contributor Guide and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [ ] I did not modify the `CHANGELOG.md` nor the `pubspec.yaml` files.
- [ ] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [ ] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by nohli** [comment]
@miquelbeltran maybe the same Windows build issue as [here](https://github.com/fluttercommunity/plus_plugins/pull/1235#issuecomment-1277924071)

File: packages/share_plus/share_plus/windows/CMakeLists.txt [hunk 1]
```diff
@@ -5,24 +5,24 @@
...
+  "share_plus_plugin.cpp"
   "share_plus_windows_plugin.h"
 )
 
 # Define the plugin library target. Its name must not be changed (see comment
 # on PLUGIN_NAME above).
 add_library(${PLUGIN_NAME} SHARED
-  "include/share_plus_windows/share_plus_windows_plugin_c_api.h"
-  "share_plus_windows_plugin_c_api.cpp"
+  "include/share_plus/share_plus_windows_plugin_c_api.h"
```
Comment (nohli) [inline packages/share_plus/share_plus/windows/CMakeLists.txt:24]: @miquelbeltran this file was deleted

Comment (miquelbeltran) [reply packages/share_plus/share_plus/windows/CMakeLists.txt:24]: forgot to add the files after rename :)

**Comment by miquelbeltran** [comment]
this windows build seems fixed now as well. I am not happy with the naming (some files have `_windows_`, some don't) but it works.

**Comment by nohli** [comment]
You're a champ 💪🏽

**Review by miquelbeltran — APPROVED** [review]

**Review by vbuberen — APPROVED** [review]

**Pull request merged**